python/ray/actor.py

```diff
@@ -1777,19 +1780,12 @@ def exit_actor():
             worker is not an actor.
     """
     worker = ray._private.worker.global_worker
-    if worker.mode == ray.WORKER_MODE and not worker.actor_id.is_nil():
-        # In asyncio actor mode, we can't raise SystemExit because it will just
-        # quit the asycnio event loop thread, not the main thread. Instead, we
-        # raise a custom error to the main thread to tell it to exit.
-        if worker.core_worker.current_actor_is_asyncio():
-            raise AsyncioActorExit()
-
-        # Set a flag to indicate this is an intentional actor exit. This
-        # reduces log verbosity.
-        raise_sys_exit_with_custom_error_message("exit_actor() is called.")
-    else:
+    if worker.mode != ray.WORKER_MODE or worker.actor_id.is_nil():
         raise TypeError(
             "exit_actor API is called on a non-actor worker, "
             f"{worker.mode}. Call this API inside an actor methods"
             "if you'd like to exit the actor gracefully."
         )
+    if not worker.core_worker.current_actor_is_asyncio():
+        raise_sys_exit_with_custom_error_message("exit_actor() is called.")
+    worker.core_worker.set_current_actor_should_exit()
```

Comment:
A little confused at why this is conditional because we should try to make the behavior as uniform as possible:

- For non-async actors, don't we still want to set the field in the core worker? Otherwise, if users accidentally or intentionally catch the `SystemExit` exception that is raised, the actor won't actually be exited.
- For async actors, don't we still want to raise the `SystemExit` exception immediately?